php/settings/class-setting.php

```diff
@@ -0,0 +1,673 @@
+				continue;
+			}
+			// Set params.
+			$this->set_param( $param, $value );
+		}
+
+		// Register dynamics.
+		$this->register_dynamic_settings( $dynamic_params );
+	}
+
+	/**
+	 * Register the setting with WordPress.
+	 */
+	protected function register_setting() {
+		$option_group = $this->get_option_name();
+		if ( ! is_null( $option_group ) && $this->get_parent() && ! $this->has_settings() && ! $this->get_parent()->has_param( 'setting_registered' ) ) {
+			$args = array(
+				'type'              => 'array',
+				'description'       => $this->get_param( 'description' ),
+				'sanitize_callback' => array( $this, 'prepare_sanitizer' ),
+				'show_in_rest'      => false,
+			);
+			register_setting( $option_group, $option_group, $args );
+			add_filter( 'pre_update_site_option_' . $option_group, array( $this, 'set_notices' ), 10, 3 );
+			add_filter( 'pre_update_option_' . $option_group, array( $this, 'set_notices' ), 10, 3 );
+			$this->get_parent()->set_param( 'setting_registered', true );
+		}
+	}
+
+	/**
+	 * Prepare the setting option group to be sanitized by each component.
+	 *
+	 * @param array $data Array of values to sanitize.
+	 *
+	 * @return array
+	 */
+	public function prepare_sanitizer( $data ) {
+		$slug = $this->get_slug();
+		if ( isset( $data[ $slug ] ) ) {
+			$data[ $slug ] = $this->get_component()->sanitize_value( $data[ $slug ] );
+		}
+
+		return $data;
+	}
+
+	/**
+	 * Set notices on successful update of settings.
+	 *
+	 * @param mixed  $value        The new Value.
+	 * @param mixed  $old_value    The old value.
+	 * @param string $setting_slug The setting key.
+	 *
+	 * @return mixed
+	 */
+	public function set_notices( $value, $old_value, $setting_slug ) {
+
+		if ( $value !== $old_value ) {
+			if ( is_wp_error( $value ) ) {
+				add_settings_error( $setting_slug, 'setting_notice', $value->get_error_message(), 'error' );
+				$value = $old_value;
+			} else {
+				$setting = $this->get_root_setting()->find_setting( $setting_slug );
+				$notice  = $setting->get_param( 'success_notice', __( 'Settings updated successfully', 'cloudinary' ) );
+				add_settings_error( $setting_slug, 'setting_notice', $notice, 'updated' );
+			}
+		}
+
+		return $value;
+	}
+
+	/**
+	 * Register dynamic settings from params.
+	 *
+	 * @param array $params Array of params to create dynamic settings from.
+	 */
+	protected function register_dynamic_settings( $params ) {
+		foreach ( $params as $param => $value ) {
+			$callback = $this->get_setting_param_callback( $param );
+			$callable = is_callable( $callback );
+			if ( $callable ) {
+				call_user_func( $callback, $value );
+			}
+		}
+	}
+
+	/**
+	 * Get a callback to handle a dynamic child setting creation.
+	 *
+	 * @param string $param Param name to get callback for.
+	 *
+	 * @return string
+	 */
+	protected function get_setting_param_callback( $param ) {
+		return $this->is_setting_param( $param ) ? $this->setting_params[ $param ] : '__return_null()';
```

Comment:
@DavidCramer 

I think there's a typo here. Shouldn't it be `__return_null`, without the parenthesis?

Reply:
Ooops!